### PR TITLE
Add test cases for JSON resource marshaling

### PR DESCRIPTION
**Pull request by sagar23sj**
Resources Covered :
- InteractionRestriction

Helps with issue: #55

### CODE REVIEW

File: github/interactions_test.go
Comment (gmlewis) [inline github/interactions_test.go:1]:
```suggestion
// Copyright 2021 The go-github AUTHORS. All rights reserved.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @sagar23sj !
One minor tweak to fix, please, and then it LGTM.
Whenever a new file is created, we use the year it was created as its copyright year.

File: github/interactions_test.go
Comment (sagar23sj) [reply github/interactions_test.go:1]: Sure @gmlewis. Will make the change.
Thank you for your instantaneous feedback.

**Review by gmlewis — APPROVED** [review]
Thank you, @sagar23sj !
LGTM.

Awaiting second LGTM before merging.

**Review by Parker77 — APPROVED** [review]
LGTM.

**Pull request merged**
